src/libNode/FinalBlockProcessing.cpp

```diff
@@ -758,6 +758,10 @@ bool Node::ProcessFinalBlockCore(const bytes& message, unsigned int offset,
         << m_mediator.m_txBlockChain.GetLastBlock().GetHeader().GetBlockNum() +
                1
         << "] FINISH WRITE STATE TO DISK");
+    if (m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() <
```

Comment:
Should we add a check for `ENABLE_ACCOUNTS_POPULATING` to make this part more efficient?

Reply:
It's checked inside the function in the first line, should not be an issue to the efficiency